autopush-common/src/db/mod.rs

```diff
@@ -161,6 +163,19 @@ pub struct User {
     /// UUID4 version number for optimistic locking of updates on Bigtable
     #[serde(skip_serializing)]
     pub version: Option<Uuid>,
+    /// Set of user's channel ids. These are stored in router (user) record's
+    /// row in Bigtable. They are read along with the rest of the user record
+    /// so that them, along with every other field in the router record, will
+    /// automatically have their TTL (cell timestamp) reset during
+    /// [DbClient::update_user].
+    ///
+    /// This is solely used for the sake of that update thus private.
+    /// [DbClient::get_channels] is preferred for reading the latest version of
+    /// the channel ids (partly due to historical purposes but also is a more
+    /// flexible API that might benefit different, non Bigtable [DbClient]
+    /// backends that don't necessarily store the channel ids in the router
+    /// record).
+    _channels: HashSet<Uuid>,
```

Comment:
The only problem I have with this is that rust has the syntax that things prefixed with an `_` are unused, not private (like python). This is already private because we don't declare it `pub`, so maybe we should go with something like `priv_channels` or `channel_cache` or something so some future person isn't confused about why we're keeping this unused variable in the structure.